### PR TITLE
Fail fast on read operations when underlying stream is closed.

### PR DESCRIPTION
**Pull request by abmodi**

### CODE REVIEW

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java [hunk 1]
```diff
@@ -193,4 +203,16 @@ public int available() throws IOException {
...
+
+  /**
+   * Verify that the input stream is open. Non blocking; this gives the last state of the volatile
+   * {@link #closed} field.
+   *
+   * @throws IOException if the connection is closed.
+   */
+  private void checkNotClosed() throws IOException {
+    if (closed) {
+      throw new IOException(gcsPath + ": " + FSExceptionMessages.STREAM_IS_CLOSED);
```
Comment (arunkumarchacko) [inline gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java:215]: should this be ClosedChannelException?

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java [hunk 2]
```diff
@@ -126,6 +134,7 @@ public synchronized int read(@Nonnull byte[] buf, int offset, int length) throws
 
   @Override
   public synchronized void seek(long pos) throws IOException {
+    checkNotClosed();
```
Comment (arunkumarchacko) [inline gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java:137]: should we add the same check for getPos() as well?

**Review by arunkumarchacko — APPROVED** [review]

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java (hunk 1 above)
Comment (abmodi) [reply gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java:215]: Yes - I tried this - it doesn't take a string as an argument. That's why I decided to go ahead with IOException.

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java (hunk 2 above)
Comment (abmodi) [reply gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java:137]: Done.

**Pull request merged**